### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by liubko**
- Update `ResponsiveContainer` / `LineChart` / `BarChart`
- add `PxOrPrc` module
- add examples
- format code with `refmt`

### CODE REVIEW

**Comment by liubko** [comment]
@rusty-key let me know what you think. I'm planning to cover more component's API here or preferably in a new PR

**Comment by liubko** [comment]
Just learned a trick from Javier need to use `_type` instead of `type_`.

Will update PR tomorrow

File: examples/src/LineChartExample.re [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  render: _self =>
+    <ResponsiveContainer height=(Px(200)) width=(Px(300))>
+      <LineChart margin={"top": 0, "right": 0, "bottom": 0, "left": 0} data>
+        <Line name="Some bar" dataKey="pv" stroke="#2078b4" />
+        <Line name="Other bar" dataKey="uv" stroke="#ff7f02" />
+        <Tooltip />
+        <Legend align="left" iconType="circle" />
+      </LineChart>
+    </ResponsiveContainer>,
+};
```
Comment (rusty-key) [inline examples/src/LineChartExample.re]: Can you please add `"files.insertFinalNewline": true,` to your editor config?

**Comment by rusty-key** [comment]
https://github.com/ahrefs/bs-recharts/pull/1/files#diff-09d6724bde495fd0c2ba310fd66df417

Let's stick with yarn in this package?

**Comment by rusty-key** [comment]
Other than that it's awesome!
By the way, do we want to cover package-specific types in documentation?

File: src/Utils.re [hunk 2]
```diff
@@ -0,0 +1,37 @@
+module PxOrPrc = {
+  type t;
+  type arg =
+    | Px(int)
```
Comment (rusty-key) [inline src/Utils.re]: I am a little bit afraid of this generic names. What will happen if there will be another package which defines same types?

**Comment by liubko** [comment]
- updated other components as well Line/Bar/CasterianGrid/Casterianaxis/Tooltip/Legend

> By the way, do we want to cover package-specific types in documentation?

- updated Readme

@rusty-key please take a look again

**Comment by Eric-Arellano** [comment]
@rusty-key @liubko these bindings are great! Thank you for making these. They're saving me a lot of time on project.

(Wasn't sure where to put this)

**Pull request merged**